apps/apps.c

```diff
@@ -273,9 +273,10 @@ int password_callback(char *buf, int bufsiz, int verify, PW_CB_DATA *cb_tmp)
 #ifdef OPENSSL_NO_UI
     if (cb_data != NULL && cb_data->password != NULL) {
         res = strlen(cb_data->password);
-        if (res > bufsiz)
-            res = bufsiz;
+        if (res > bufsiz - 1)
+            res = bufsiz - 1;
         memcpy(buf, cb_data->password, res);
+        buf[res] = 0;
```

Comment:
@levitte, it's a left over from the previous incarnation. That commit needs to be removed from this PR